paimon-core/src/main/java/org/apache/paimon/partition/PartitionExpireStrategy.java

```diff
@@ -49,6 +50,14 @@ public Map<String, String> toPartitionString(Object[] array) {
         return map;
     }
 
+    public List<String> toPartitionValue(Object[] array) {
+        List<String> list = new ArrayList<>();
```

Comment:
Change "List<String> list = new ArrayList<>();" to "List<String> list = new ArrayList<>(partitionKeys.size());".